src/OrchardCore.Modules/OrchardCore.Lucene/Services/SearchQueryService.cs

```diff
@@ -21,15 +21,16 @@ public async Task<IList<string>> ExecuteQueryAsync(Query query, string indexName
 
             await _luceneIndexManager.SearchAsync(indexName, searcher =>
             {
-                var collector = TopScoreDocCollector.Create(end, true);
-
-                searcher.Search(query, collector);
-                var hits = collector.GetTopDocs(start, end);
-
-                foreach (var hit in hits.ScoreDocs)
+                if (end > 0)
```

Comment:
How do you get `end` to be `0` ?

Reply:
When there is no index files found

Reply:
This PR needs more work. I'm not sure this is the proper fix. It's a nice patch but that does not fix the underlying issue. If you expect to be able to search in an index there should be an index available.

Reply:
Example : 

```JSON
{
  "from" : 0,
  "size": 0,
  "query": {
    "match_all": { }
  }
}
```